packages/core/mesh/network-manager/src/transport/webrtc-transport.ts

```diff
@@ -1,129 +1,34 @@
- */
-export class WebRTCTransport implements Transport {
-  private readonly _instanceId = PublicKey.random().toHex();
-  private readonly _peer: SimplePeer;
-
-  private _closed = false;
-  readonly closed = new Event();
-  readonly connected = new Event();
-  readonly errors = new ErrorStream();
-
-  constructor(private readonly params: WebRTCTransportParams) {
-    log.trace('dxos.mesh.webrtc-transport.constructor', trace.begin({ id: this._instanceId }));
-    log('created connection', params);
-
-    // https://www.npmjs.com/package/simple-peer#api
-    // NOTE: Set localStorage.debug = 'simple-peer*' to enable debug logging. (Or show verbose in Chrome).
-    // View throughput via: chrome://webrtc-internals (data-channel)
-    this._peer = new SimplePeerConstructor({
-      initiator: this.params.initiator,
-      config: this.params.webrtcConfig,
-      // https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createDataChannel
-      channelConfig: {
-        protocol: 'dxos.mesh',
-      },
-      // Custom WebRTC implementation (for Node); otherwise get-browser-rtc.
-      wrtc: SimplePeerConstructor.WEBRTC_SUPPORT ? undefined : wrtc ?? raise(new Error('wrtc not available')),
-    });
-
-    this._peer.on('signal', async (data) => {
-      log('signal', data);
-      await this.params.sendSignal({ payload: { data } });
-    });
-
-    this._peer.on('connect', () => {
-      log('connected');
-      this.params.stream.pipe(this._peer!).pipe(this.params.stream);
-      this.connected.emit();
-    });
-
-    this._peer.on('close', async () => {
-      log('closed');
-      await this._disconnectStreams();
-      this.closed.emit();
-    });
-
-    this._peer.on('error', async (err) => {
-      this.errors.raise(err);
-
-      // Try to gather additional information about the connection.
-      try {
-        if (typeof (this._peer as any)?._pc.getStats === 'function') {
-          (this._peer as any)._pc.getStats().then((stats: any) => {
-            log.warn('report after webrtc error', {
-              config: this.params.webrtcConfig,
-              stats,
-            });
-          });
-        }
-      } catch {} // TODO(burdon): Ignored?
-
-      await this.destroy();
-    });
-    log.trace('dxos.mesh.webrtc-transport.constructor', trace.end({ id: this._instanceId }));
-  }
-
-  async destroy() {
-    log('closing...');
-    this._closed = true;
-    await this._disconnectStreams();
-    this._peer!.destroy();
-    this.closed.emit();
-    log('closed');
-  }
-
-  signal(signal: Signal) {
-    if (this._closed) {
-      return; // Ignore signals after close.
+import { LibDataChannelTransport } from './libdatachannel-transport';
+import { SimplePeerTransport } from './simplepeer-transport';
+import { TransportFactory } from './transport';
+
+export const createWebRTCTransportFactory = (webrtcConfig?: any, webrtcLibrary = 'SimplePeer'): TransportFactory => {
```

Comment:
split those into two separate factories; you could move them into the files where the classes for transports are defined